api/proto/config/storage.proto

```diff
@@ -193,3 +193,32 @@ message Volume {
   string displayName = 7;  // Optional friendly name echo'ed in info message
   bool clear_text = 8;  // Flag to indicate the volume encryption needed or not
 }
+
+// DiskConfigType is the desired configuration of disks
+enum DiskConfigType {
+  DISK_CONFIG_TYPE_UNSPECIFIED = 0; // no configured, do nothing
+  DISK_CONFIG_TYPE_EVEOS = 1; // the disk EVE is installed on
+  DISK_CONFIG_TYPE_PERSIST = 2; // the disk is separate persist partition, not zfs
+  DISK_CONFIG_TYPE_ZFS_ONLINE = 3; // included in zfs and online
+  DISK_CONFIG_TYPE_ZFS_OFFLINE = 4; // included in zfs and offline
```

Comment:
What about included in ZFS but faulted/removed etc. Do we consider all those states as OFFLINE ?

Reply:
Here I want to define state that we want to move the device onto. Something like "make disk_name online/offline". So I not expect to see here something like "make disk_name_faulted".
Remove is going out from zfs_online/offline

Reply:
Oh ok these are not used for current state of ZFS devices. Really for unconfigured/unused disks on the system and then later move them accordingly ? Makes sense, but we configure ZFS during installation itself, so adding these states will be useful only for new additional disks that might be added to ZFS ?

Reply:
If we will find disk which is already in pool here, we will check and modify its state if needed. If we will find disk which is not in pool, we will try to work with it (with support of pool expansion, we will expand, right now we can think about replacement). If we will not see anything about disks, we will do nothing.

Reply:
ok got it.